src/ProgressBar/index.jsx

```diff
@@ -1 +1,147 @@
+const VARIANTS = [
+  'dark',
+  'warning',
+  'success',
+  'error',
+];
+
+const ProgressBar = (props) => <ProgressBarBase {...props} />;
+
+const ProgressBarAnnotated = ({
+  now,
+  label,
+  variant,
+  threshold,
+  thresholdLabel,
+  thresholdVariant,
+  progressHint,
+  thresholdHint,
+  ...props
+}) => {
+  const progressInfoRef = React.useRef();
+  const thresholdInfoRef = React.useRef();
+  const thresholdPercent = (threshold || 0) - (now || 0);
+  const isProgressHintAfter = now < HINT_SWAP_PERCENT;
+  const isThresholdHintAfter = threshold < HINT_SWAP_PERCENT;
+  const progressColor = VARIANTS.includes(variant) ? variant : PROGRESS_DEFAULT_VARIANT;
+  const thresholdColor = VARIANTS.includes(thresholdVariant) ? thresholdVariant : THRESHOLD_DEFAULT_VARIANT;
+
+  useEffect(() => {
+    placeInfoAtZero(progressInfoRef, isProgressHintAfter, ANNOTATION_CLASS);
+    placeInfoAtZero(thresholdInfoRef, isThresholdHintAfter, ANNOTATION_CLASS);
+  }, []);
+
+  const getHint = (text) => (
+    <span className="pgn__progress-hint">
+      {text}
+    </span>
+  );
+
+  return (
+    <div className="pgn__progress-annotated">
+      {!!label && (
+        <div
+          className="pgn__progress-info"
+          style={{ left: `${now}%` }}
+          ref={progressInfoRef}
+        >
+          {!isProgressHintAfter && getHint(progressHint)}
+          <Annotation variant={progressColor}>
+            {label}
+          </Annotation>
+          {isProgressHintAfter && getHint(progressHint)}
+        </div>
+      )}
+      <ProgressBarBase>
+        <ProgressBarBase
+          {...props}
+          now={now}
+          className={classNames(
+            `pgn__progress-bar--${progressColor}`,
+            thresholdPercent > 0 ? 'pgn__progress-tick--white' : 'pgn__progress-tick--black',
+          )}
+          srOnly
+        />
+        {!!threshold && (
+          <ProgressBarBase
+            now={thresholdPercent}
+            className={`pgn__progress-bar--${thresholdColor}`}
+            srOnly
+          />
+        )}
+        {!!threshold && (
+          <div
+            className={`pgn__progress-threshold-dot--${thresholdColor}`}
+            style={{ left: `${threshold}%` }}
+          />
+        )}
+      </ProgressBarBase>
+      {(!!threshold && !!thresholdLabel) && (
+        <div
+          className="pgn__progress-info"
+          style={{ left: `${threshold}%` }}
+          ref={thresholdInfoRef}
+        >
+          {!isThresholdHintAfter && getHint(thresholdHint)}
+          <Annotation
+            arrowPlacement="top"
+            variant={thresholdColor}
+          >
+            {thresholdLabel}
+          </Annotation>
+          {isThresholdHintAfter && getHint(thresholdHint)}
+        </div>
+      )}
+    </div>
+  );
+};
+
+ProgressBarAnnotated.propTypes = {
+  /** Current value of progress. */
+  now: PropTypes.number,
+  /** Show label that represents visual percentage. */
+  label: PropTypes.string,
```

Comment:
I believe this could also be PropTypes.node so that consumers are able to pass in an element/component if they need to, too. Example:

```jsx
  <ProgressBar.Annotated
    now={20}
    label={<span data-testid="progressbar-label">20%</span>}
    {/* ... */}
/>
```

Similarly for `thresholdLabel`, `progressHint`, `thresholdHint`, too.

Reply:
Changed `label`, `thresholdLabel`, `progressHint`, `thresholdHint` to `PropTypes.node` 👍